smartconfig/entry.py

```diff
@@ -66,10 +66,14 @@ def _register_entry(cls) -> None:
         _registry.configuration_for_module[cls.__path] = cls
 
     def _check_undefined_entries(cls) -> None:
-        """Raise `ConfigurationKeyError` if any attribute doesn't have a concrete value."""
+        """Raise `ConfigurationKeyError` if any attribute doesn't have a defined value."""
         for attribute in cls.__defined_attributes:
-            if not hasattr(cls, attribute) and attribute not in _registry.global_configuration[cls.__path]:
-                raise ConfigurationKeyError(f"Attribute {attribute!r} isn't defined.")
+            try:
+                if hasattr(cls, attribute):
+                    continue
+            except ConfigurationKeyError:
```

Comment:
This doesn't seem right. `hasattr` shouldn't raise an error for just trying to check if an attribute exists. The reason is likely because you're raising a custom exception instead of `AttributeError`, which is what `hasattr`'s implementation expects to catch. I believe I suggested previously that this exception should also subclass `AttributeError`.